refaster-support/src/test/java/tech/picnic/errorprone/refaster/matchers/IsIdentityOperationTest.java

```diff
@@ -0,0 +1,73 @@
+package tech.picnic.errorprone.refaster.matchers;
+
+import static com.google.errorprone.BugPattern.SeverityLevel.ERROR;
+
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.CompilationTestHelper;
+import com.google.errorprone.bugpatterns.BugChecker;
+import org.junit.jupiter.api.Test;
+
+final class IsIdentityOperationTest {
+  @Test
+  void matches() {
+    CompilationTestHelper.newInstance(MatcherTestChecker.class, getClass())
+        .addSourceLines(
+            "A.java",
+            "import java.util.function.BinaryOperator;",
+            "import java.util.function.DoubleUnaryOperator;",
+            "import java.util.function.Function;",
+            "import java.util.function.IntUnaryOperator;",
+            "import java.util.function.LongUnaryOperator;",
+            "import java.util.function.UnaryOperator;",
+            "",
+            "class A {",
+            "  BinaryOperator<String> negative1() {",
+            "    return (a, b) -> a;",
+            "  }",
+            "",
+            "  DoubleUnaryOperator positive1() {",
+            "    // BUG: Diagnostic contains:",
+            "    return DoubleUnaryOperator.identity();",
+            "  }",
+            "",
+            "  Function<Integer, Integer> positive2() {",
+            "    // BUG: Diagnostic contains:",
+            "    return Function.identity();",
+            "  }",
+            "",
+            "  UnaryOperator<String> positive3() {",
+            "    // BUG: Diagnostic contains:",
+            "    return UnaryOperator.identity();",
+            "  }",
+            "",
+            "  IntUnaryOperator positive4() {",
+            "    // BUG: Diagnostic contains:",
+            "    return IntUnaryOperator.identity();",
+            "  }",
+            "",
+            "  LongUnaryOperator positive5() {",
+            "    // BUG: Diagnostic contains:",
+            "    return LongUnaryOperator.identity();",
+            "  }",
```

Comment:
Previously we would only match `Function.identity()`. Our internal code base contains a number of `UnaryOperator.identity()` usages that were thus not matched by the Refaster rules.